apps/condo/domains/miniapp/schema/B2CAppProperty.js

```diff
@@ -0,0 +1,115 @@
+    labelResolver: async (item) => {
+        const app = await getById('B2CApp', item.app)
+        const appName = get(app, 'name', 'deleted')
+        const appDeveloper = get(app, 'developer', 'deleted')
+        return `${appDeveloper}-${appName}-${item.address}`
+    },
+    fields: {
+        app: {
+            schemaDoc: 'Link to B2C App',
+            type: Relationship,
+            ref: 'B2CApp.properties',
+            isRequired: false,
+            kmigratorOptions: { null: true, on_delete: 'models.PROTECT' },
+            access: {
+                update: access.canUpdateAppLink,
+            },
+            hooks: {
+                validateInput: async ({ operation, resolvedData, addFieldValidationError }) => {
+                    if (operation === 'update') {
+                        if (resolvedData.hasOwnProperty('app') && resolvedData.app !== null) {
+                            return addFieldValidationError(RESTRICT_PROPERTY_APP_CHANGE_ERROR)
+                        }
+                    }
+                },
+            },
+        },
+        address: {
+            schemaDoc: 'Property address. Must match the address lookup service with case accuracy',
+            type: Text,
+            isRequired: true,
+            hooks: {
+                resolveInput: ({ resolvedData, fieldPath }) => {
+                    if (typeof resolvedData[fieldPath] === 'string') {
+                        resolvedData[fieldPath] = resolvedData[fieldPath].toLowerCase()
+                    }
+                    return resolvedData[fieldPath]
+                },
+                validateInput: async ({ resolvedData, fieldPath, addFieldValidationError }) => {
+                    const inputAddress = resolvedData[fieldPath]
+                    if (!inputAddress) {
+                        return addFieldValidationError(INCORRECT_ADDRESS_ERROR)
+                    }
+                    // TODO(DOMA-3250) Fix this mess after migration to new address-suggestion service
+                    const suggestions = await getAddressSuggestions(inputAddress, 1)
+                    const suggestionAddress = get(suggestions, ['0', 'value'], '')
+                    const suggestionHouseType = get(suggestions, ['0', 'data', 'house_type_full'])
+                    if (!VALID_HOUSE_TYPES.includes(suggestionHouseType)) {
+                        return addFieldValidationError(`${INCORRECT_HOUSE_TYPE_ERROR}. Valid values are: [${VALID_HOUSE_TYPES.join(', ')}]`)
+                    }
+                    if (suggestionAddress.toLowerCase() !== inputAddress.toLowerCase()) {
+                        return addFieldValidationError(INCORRECT_ADDRESS_ERROR)
+                    }
+                },
+            },
+        },
+    },
+    hooks: {
+        resolveInput: async ({ resolvedData }) => {
+            if (resolvedData && resolvedData.hasOwnProperty('app') && resolvedData['app'] === null) {
```

Comment:
it look like a hack for admin interface restrictions

Reply:
removed